odoo/addons/base/static/src/js/res_config_settings.js

```diff
@@ -344,22 +346,26 @@ var BaseSettingRenderer = FormRenderer.extend({
             this._resetSearch();
         }
     },
+
     /**
      * highlight search word
      *
      * @private
-     * @param {string} text
+     * @param {HTMLElement} node
      * @param {string} word
      */
-    _wordHighlighter: function (text, word) {
-        if (text.indexOf('highlighter') !== -1) {
-            text = text.replace('<span class="highlighter">', "");
-            text = text.replace("</span>", "");
-        }
-        var match = text.search(new RegExp(word, "i"));
-        word = text.substring(match, match + word.length);
-        var highlightedWord = "<span class='highlighter'>" + word + '</span>';
-        return text.replace(word, highlightedWord);
+    _wordHighlighter: function (node, word) {
+        const text = node.innerText;
+        $(node).empty();
+
+        const startIndex = text.toUpperCase().indexOf(word.toUpperCase());
+        const endIndex = startIndex + word.length;
+
+        $(node).append(
+            text.substring(0, startIndex),
+            $('<span class="highlighter">').text(text.substring(startIndex, endIndex)),
```

Comment:
```suggestion
            $('<span class="highlighter">', {text: text.substring(startIndex, endIndex)}),
```
might be clearer? I'm always of two minds about that.

Reply:
Uh nm, I tested a simple case (without a class) and since it worked fine I assumed it worked fine always, but apparently it can't be combined with "inline" classes, it's either everything in the node or everything in the attributes:
```js
>> $('<span>', {text: "test"})[0].outerHTML
"<span>test</span>"
>> $('<span class="highlighter">', {text: "test"})[0].outerHTML
"<span class=\"highlighter\"></span>"
>> $('<span>', {text: "test", className: "highlighter"})[0].outerHTML
"<span classname=\"highlighter\">test</span>"
```
Sorry about that.